tests/end2end/exporter.py

```diff
@@ -0,0 +1,97 @@
+import os
+import shutil
+import subprocess
+import tempfile
+from pathlib import Path
+from time import sleep
+from typing import Optional
+
+from strictdoc.helpers.file_system import get_portable_temp_dir
+from tests.end2end.conftest import test_environment
+
+
+class SDocTestHTMLExporter:
+    def __init__(
+        self,
+        *,
+        input_path: str,
+        output_path: Optional[str] = None,
+        config_path: Optional[str] = None,
+    ):
+        assert os.path.isdir(input_path)
+        if config_path is not None:
+            assert os.path.exists(config_path)
+        if output_path is not None:
+            assert os.path.exists(output_path)
+            self.cleanup_output_path = False
+        else:
+            output_path = tempfile.mkdtemp()
+            self.cleanup_output_path = True
+
+        self.path_to_tdoc_folder = input_path
+        self.output_path: Optional[str] = output_path
+        self.config_path: Optional[str] = config_path
+
+        path_to_tmp_dir = get_portable_temp_dir()
+        self.path_to_out_log = os.path.join(
+            path_to_tmp_dir, "strictdoc_export.out.log"
+        )
+        self.path_to_err_log = os.path.join(
+            path_to_tmp_dir, "strictdoc_export.err.log"
+        )
```

Comment:
Here somewhere `self.process = None` is missing and also let's add a type annotation to all variables.